packages/client/src/RpcProviderFactory.ts

```diff
@@ -18,10 +22,16 @@ export class RpcProviderFactory {
             // eslint-disable-next-line no-underscore-dangle
             const timeout = this.config._timeouts.jsonRpcTimeout
             const pollInterval = this.config.contracts.pollInterval
-            this.providers = this.config.contracts.streamRegistryChainRPCs.rpcs.map((c: ConnectionInfo) => {
-                const provider = new LoggingStaticJsonRpcProvider({
-                    ...c,
-                    timeout
+            this.providers = this.config.contracts.streamRegistryChainRPCs.rpcs.map((c) => {
+                const fetchRequest = new FetchRequest(c.url)
+                fetchRequest.timeout = timeout
+                const provider = new LoggingJsonRpcProvider(fetchRequest, {
+                    chainId: this.config.contracts.streamRegistryChainRPCs?.chainId,
+                    name: this.config.contracts.streamRegistryChainRPCs?.name
```

Comment:
Maybe `chainId` and `name` fields should be required properties, if `LoggingJsonRpcProvider` doesn't work ok without those values? (Not sure if that was the case)